src/stdlib/format_timestamp.rs

```diff
@@ -43,39 +53,66 @@ impl Function for FormatTimestamp {
     ) -> Compiled {
         let value = arguments.required("value");
         let format = arguments.required("format");
+        let tz: Option<Box<dyn Expression>> = arguments.optional("tz");
 
-        Ok(FormatTimestampFn { value, format }.as_expr())
+        Ok(FormatTimestampFn { value, format, tz }.as_expr())
     }
 
     fn examples(&self) -> &'static [Example] {
-        &[Example {
-            title: "format timestamp",
-            source: r#"format_timestamp!(t'2021-02-10T23:32:00+00:00', "%d %B %Y %H:%M")"#,
-            result: Ok("10 February 2021 23:32"),
-        }]
+        &[
+            Example {
+                title: "format timestamp",
+                source: r#"format_timestamp!(t'2021-02-10T23:32:00+00:00', format: "%d %B %Y %H:%M")"#,
+                result: Ok("10 February 2021 23:32"),
+            },
+            Example {
+                title: "format timestamp with tz",
+                source: r#"format_timestamp!(t'2021-02-10T23:32:00+00:00', format: "%d %B %Y %H:%M", tz: "Europe/Berlin")"#,
+                result: Ok("11 February 2021 00:32"),
+            },
+        ]
     }
 }
 
 #[derive(Debug, Clone)]
 struct FormatTimestampFn {
     value: Box<dyn Expression>,
     format: Box<dyn Expression>,
+    tz: Option<Box<dyn Expression>>,
 }
 
 impl FunctionExpression for FormatTimestampFn {
     fn resolve(&self, ctx: &mut Context) -> Resolved {
         let bytes = self.format.resolve(ctx)?;
         let ts = self.value.resolve(ctx)?;
 
-        format_timestamp(bytes, ts)
+        match self.tz.clone() {
+            None => format_timestamp_with_tz(bytes, ts, &Utc),
+            Some(tz) => {
+                let tz = &tz.resolve(ctx)?.try_bytes()?;
+                let tz = String::from_utf8_lossy(tz);
+                match tz {
+                    std::borrow::Cow::Borrowed("Local") => {
+                        format_timestamp_with_tz(bytes, ts, &Local)
+                    }
+                    _ => {
+                        let tz: Tz = tz.parse().unwrap();
```

Comment:
Rather than `unwrap()` I think we'll want to return the error here. I think you should just be able to return a string like:

```suggestion
                        let tz: Tz = tz.parse().map_err(|err| format!("unable to parse timezone: {}", err))?;
```

Reply:
Applied